tests/utilities/test_distributed.py

```diff
@@ -64,15 +64,14 @@ def foo():
 def _test_collect_states(rank, world_size):
     os.environ["MASTER_ADDR"] = "localhost"
 
+    torch.cuda.set_device(f"cuda:{rank}")
+
```

Comment:
Why is this required now?

Reply:
I believe there is a bug with `torch.distributed.broadcast_object_list` or it might be a design choice.

But [here](https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L1855) they rely on torch.cuda.current_device() instead of torch.distributed.get_rank() meaning both tensors are still on `cuda:0` and the broadcast would fail.